### PR TITLE
V0.9

### PR DESCRIPTION
**Pull request by gerardofn**
- Support for ARM
- New API layer to facilitate porting to other platforms/tools

### CODE REVIEW

File: plugin/virustotal/vtgrep.py [hunk 1]
```diff
@@ -119,6 +123,11 @@ def pack(self):
         self.wcs_stream += '?' * (wcs_len % 2)
         self.packed = True
 
+  def same_type(self, next_slice):
+    if isinstance(next_slice, Bytes):
```
Comment (winden-g) [inline plugin/virustotal/vtgrep.py]: Here, should be Wilcards?

File: plugin/virustotal/vt_ida/disassembler.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  @staticmethod
+  def valid_address_range(addr_start, addr_end):
+    if addr_start == idaapi.BADADDR or addr_end == idaapi.BADADDR:
+      return False
+    else:
+      return True
+
+  @staticmethod
+  def valid_range_size(addr_start, addr_end, max_size):
+    if (addr_end - addr_start) > max_size:
```
Comment (winden-g) [inline plugin/virustotal/vt_ida/disassembler.py]: return (end - start) <= max:

File: plugin/virustotal/vt_ida/disassembler.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import idautils
+import idc
+import logging
+
+
+class Disassembler(object):
+
+  @staticmethod
+  def valid_address_range(addr_start, addr_end):
+    if addr_start == idaapi.BADADDR or addr_end == idaapi.BADADDR:
```
Comment (winden-g) [inline plugin/virustotal/vt_ida/disassembler.py]: if start == self.bad_address() or ...

File: plugin/virustotal/vt_ida/disassembler.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+import idaapi
+import idautils
+import idc
+import logging
+
+
+class Disassembler(object):
+
+  @staticmethod
+  def valid_address_range(addr_start, addr_end):
```
Comment (winden-g) [inline plugin/virustotal/vt_ida/disassembler.py]: start_addr, end_addr (here and elsewhere)

File: plugin/virustotal/vt_ida/widgets.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2019 Google Inc. All Rights Reserved.
```
Comment (winden-g) [inline plugin/virustotal/vt_ida/widgets.py]: 2020

File: plugin/virustotal/vt_ida/disassembler.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
+# Copyright 2019 Google Inc. All Rights Reserved.
```
Comment (winden-g) [inline plugin/virustotal/vt_ida/disassembler.py]: 2020

File: plugin/virustotal/vtgrep.py [hunk 7]
```diff
@@ -14,12 +14,11 @@
 __author__ = 'gerardofn@virustotal.com'
 
 import binascii
-import ida_kernwin
-import idaapi
-import idautils
-import idc
 import logging
+from virustotal.vt_ida.disassembler import *
```
Comment (winden-g) [inline plugin/virustotal/vtgrep.py]: import * is a fairly bad pattern -- please import each class or function explicitly

File: plugin/vt.py [hunk 8]
```diff
@@ -14,9 +14,7 @@
 
 __author__ = 'gerardofn@virustotal.com'
 
-import idaapi
-
-from vt_ida.plugin_loader import *
+from virustotal.vt_ida.plugin_loader import *
```
Comment (winden-g) [inline plugin/vt.py]: also here

File: plugin/virustotal/vtgrep.py (hunk 1 above)
Comment (gerardofn) [reply plugin/virustotal/vtgrep.py]: Not really, if the if statement is True then returns False

**Comment by gerardofn** [comment]
All comments han been resolved. Ready to merge.

**Pull request merged**